libnmstate/netapplier.py

```diff
@@ -119,6 +120,22 @@ def generate_ifaces_metadata(ifaces_desired_state, ifaces_current_state):
     )
 
 
+def sanitize_ethernet_state(ifaces_desired_state):
+    """
+    If auto-negotiation is enabled and speed and duplex settings are not
+    provided, they need to be set to None to not override them with the values
+    from the current settings. This makes it possible to distiguish whether a
+    user specified these values in the later configuration step.
+    """
+    for interface in ifaces_desired_state:
```

Comment:
for iface_state in six.viewvalues(ifaces_desired_state)

Reply:
Why is `viewvalues()` needed here?

Reply:
Thank you, good idea. I will change this.